public/js/services/cart-service.js

```diff
@@ -8,6 +8,35 @@ class CartService {
       ? JSON.parse(localStorage.getItem('cart')).length
       : 0
   }
+
+  isCartEmpty() {
+    return !localStorage.getItem('cart') ? true : false
+  }
+
+  addToCart(id, quantity) {
+    if (this.isCartEmpty()) {
+      localStorage.setItem('cart', JSON.stringify([{ id, quantity }]))
+    } else {
+      var cart = this.getCartProducts()
+      var index = cart.findIndex((x) => x.id == id)
+      if (index !== -1) {
+        cart[index].quantity = cart[index].quantity + quantity
+      } else {
+        cart.push({ id, quantity })
+      }
+      localStorage.setItem('cart', JSON.stringify(cart))
+    }
+    this.updateCart()
+  }
```

Comment:
Maybe we can move `get`ing and `save`ing the cart to separate functions to be more SRP

```js
getCartProducts() {
  return JSON.parse(localStorage.getItem('cart') || '[]')
}

saveCartProducts(cart) {
  localStorage.setItem('cart', JSON.stringify(cart))
}
```
Then we can simplify
```suggestion
  addToCart(id, quantity) {
    var cart = this.getCartProducts()
    var index = cart.findIndex((x) => x.id == id)
    if (index !== -1) {
      cart[index].quantity = cart[index].quantity + quantity
    } else {
      cart.push({ id, quantity })
    }
    this.saveCartProducts(cart)
  }
```